mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/impl/MongoDBDatastoreProvider.java

```diff
@@ -51,9 +52,12 @@
 	private ServiceRegistryImplementor serviceRegistry;
 
 	private MongoClient mongo;
-	private MongoDatabase mongoDb;
+	private volatile MongoDatabase mongoDb;
 	private MongoDBConfiguration config;
 
+	private boolean lazyGetDatabase;
```

Comment:
I don't think this boolean is necessary? It seems to only add complexity. In which case would I not want to use lazy initialization? It there are benefits in using it, let's always use it.

Reply:
don't know, maybe I want to successfully start my dependent service only if there is db connection

Reply:
Testing in this kind of situation is quite easy, one can just run a query. I don't think setting an option on/off is necessary. Let's keep it simple and remove the property for now. We can add it later if somebody ask or there is a compelling case.

Reply:
Ok, I will do the changes and open a JIRA. For the test I'll have to do some more digging.